### PR TITLE
feat(examples): Introduce Skipper as example

### PR DESCRIPTION
**Pull request by razvand**
Add Skipper as binary compatibility run. Build Skipper as static PIE application using `Dockerfile`. Then run it with the base image from the registry.

### CODE REVIEW

**Review by StefanJum — APPROVED** [review]
This works fine, thank you.
Reviewed-by: Stefan Jumarea <stefanjumarea02@gmail.com>
Approved-by: Stefan Jumarea <stefanjumarea02@gmail.com>

**Comment by razvand** [comment]
> This works fine, thank you. Reviewed-by: Stefan Jumarea [stefanjumarea02@gmail.com](mailto:stefanjumarea02@gmail.com) Approved-by: Stefan Jumarea [stefanjumarea02@gmail.com](mailto:stefanjumarea02@gmail.com)

Thanks. Let's wait for @Mihnea0Firoiu 's input as well. And I'm also looking into doing some script updates.

**Comment by razvand** [comment]
@StefanJum , @Mihnea0Firoiu , I did updates to the `utils/bincompat` PR (#8). And rebased this PR on top of that. Test this PR to also test PR #8 . Then PR #8 needs to be approved / merged first. And then this PR.

File: examples/README.md [hunk 1]
```diff
@@ -0,0 +1,106 @@
...
+
+1. Start a [BuildKit](https://docs.docker.com/build/buildkit/) container to be used by KraftKit to build the filesystem from the Dockerfile.
+   Source the corresponding script:
+
+   ```console
+   source ../../utils/start-buildkit.sh
+   ```
+
+1. Running the application requires kernel images (called `base` kernel images).
+   If not already generated, generate the kernels in `../../kernels` by running, while inside the `../../library/base/directory/`, the command:
```
Comment (StefanJum) [inline examples/README.md:46]:
```suggestion
   If not already generated, generate the kernels in `../../kernels` by running, while inside the `../../library/base/` directory, the command:
```

File: examples/README.md [hunk 2]
```diff
@@ -0,0 +1,106 @@
...
+
```
Comment (StefanJum) [inline examples/README.md:47]: Add the `kraft login` step here.

**Review by StefanJum — CHANGES_REQUESTED** [review]
See the comments.

Besides those, the scripts using `qemu` run fine, but when trying to run it using `kraft` I get `could not find runtime: too many options` (both the `kraft run ...` from the README and the generated `run-kraft` script).

**Review by Mihnea0Firoiu — COMMENTED** [review]
**./run-qemu-x86_64.sh** works as expected, although for some reason the curl doesn't work (probably a problem from skipper).
When I run **./kraft-run-qemu.sh** I get the following output:
```
dreamy_flirt
virbr0
 i  pulling index.unikraft.io/unikraft.org/base:latest
 i  #1 [internal] load build definition from Dockerfile
 i  #1 transferring dockerfile: 678B done
 i  #1 DONE 0.0s
 i  
 i  #2 [internal] load metadata for docker.io/library/golang:1.21.4-bookworm
 i  #2 DONE 0.6s
 i  
 i  #3 [internal] load .dockerignore
 i  #3 transferring context: 2B done
 i  #3 DONE 0.0s
 i  
 i  #4 [internal] load build context
 i  #4 transferring context: 35B done
 i  #4 DONE 0.0s
 i  
 i  #5 [build 1/6] FROM docker.io/library/golang:1.21.4-bookworm@sha256:52362e252f452df17c24131b021bf2ebf1c9869f65c28f88ddb326191defea9c
 i  #5 resolve docker.io/library/golang:1.21.4-bookworm@sha256:52362e252f452df17c24131b021bf2ebf1c9869f65c28f88ddb326191defea9c done
 i  #5 DONE 0.0s
 i  
 i  #6 [stage-1 1/2] COPY --from=build /usr/local/bin/skipper /usr/local/bin/skipper
 i  #6 CACHED
 i  
 i  #7 [build 5/6] RUN CGO_ENABLED=1 go build -buildmode=pie -ldflags "-linkmode external -extldflags '-static-pie'" -o /usr/local/bin/skipper ./cmd/skipper
 i  #7 CACHED
 i  
 i  #8 [build 3/6] RUN set -xe;     wget -O skipper.zip "https://github.com/zalando/skipper/archive/refs/tags/v0.18.51.zip";     unzip -q skipper.zip;
 i  #8 CACHED
 i  
 i  #9 [build 4/6] WORKDIR /skipper-0.18.51
 i  #9 CACHED
 i  
 i  #10 [build 2/6] RUN set -xe; apt-get update ;     apt-get install -y --no-install-recommends       build-essential       git       unzip       ;
 i  #10 CACHED
 i  
 i  #11 [stage-1 2/2] COPY example.eskip /etc/skipper/example.eskip
 i  #11 CACHED
 i  
 i  #12 exporting to client directory
 i  #12 copying files 47.49MB 0.1s
 i  #12 copying files 69.55MB 0.1s done
 i  #12 DONE 0.1s
 i  
 i  #13 exporting cache to client directory
 i  #13 preparing build cache for export 0.0s done
 i  #13 writing layer sha256:852d4dd247e89fbe2e5d8c7b815240a2819a90511349b1dd54a609705d308ed0 done
 i  #13 writing layer sha256:a802af1a0cfbeeeb6c54c80e7783f14e0360aea2c4bda6c767fa58d6b0a87b8c done
 i  #13 writing config sha256:b6b6d359ff48f27cbd180c352edd1328b645a53a19d3df2e9a52045f821ba23e done
 i  #13 writing cache manifest sha256:6dff306ebaaba7e419ad7e1cc9ada3a202ccda151f2dc4960d23b107e82dbffd done
 i  #13 DONE 0.0s
```
After a few seconds: ```E  machine fatally exited```

**kraft run...** doesn't work at all, with the output: ``` E  could not find runtime: too many options```

**kraft version**: ```kraft 0.7.0 (b4c61f166ade831a725e7c16edebc810f7c4329a) go1.21.2 2023-11-08T02:09:50Z```

File: examples/skipper-0.18/example.eskip [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+        -> "http://172.32.255.1/null";
+
+gold:
+        Path("/gold")
+        -> "http://172.32.255.1/gold";
+
+oneb:
+        Path("/1b")
+        -> "http://172.32.255.1/1b";
+
```
Comment (StefanJum) [inline examples/skipper-0.18/example.eskip:18]: Any reason to leave these routes here? They don't work anyway (you get an `Unhandled page fault` if you try to curl them).

**Review by StefanJum — CHANGES_REQUESTED** [review]
Looks fine, see the one comment.

**Pull request closed without merging**